### PR TITLE
Split email client

### PR DESCRIPTION
**Pull request by andrewconner**
This splits the PostOffice to two (separate, non-inheriting) classes: RemotePostOffice and LocalPostOffice. This makes usage very explicit.

As a part of this, it affected many places including nearly every module, healthchecks (requiring a new plugin / actor), etc.

### CODE REVIEW

**Comment by andrewconner** [comment]
Closing for now.

**Comment by gmethvin** [comment]
@andrewconner I've had issues with using `config.resource` in the past. Did you try using `config.file`? I have no idea why I've been able to get it to work with one and not the other.

**Comment by andrewconner** [comment]
@gmethvin I did try `config.file` as well, but on application start, it complained about unresolved unrelated dependencies (something with S3), as if it was trying to start with _no_ config file at all. I'm fairly certain the config path was correct, because I used an absolute path with `pwd`.

**Comment by gmethvin** [comment]
Oh, and I also already created the config files you're talking about at `application-shoebox.conf` and `application-search.conf`. Not sure how they differ from yours.

File: shoebox/app/com/keepit/controllers/shoebox/ShoeboxController.scala [hunk 1]
```diff
@@ -29,6 +29,10 @@ class ShoeboxController @Inject() (
   clickingHistoryRepo: ClickHistoryRepo,
   normUriRepo: NormalizedURIRepo)
   extends ShoeboxServiceController with Logging {
+  
+  def sendMail = Action(parse.json) { request =>
+    Ok("true")
```
Comment (eishay) [inline shoebox/app/com/keepit/controllers/shoebox/ShoeboxController.scala]: ?
so its not functional yet?

Comment (andrewconner) [inline shoebox/app/com/keepit/controllers/shoebox/ShoeboxController.scala]: Haha whoops, forgot to commit that. Incoming.

**Comment by eishay** [comment]
wow, a big one!

**Pull request merged**